samples/qa-virtualgridlist/src/context/RecordContext.js

```diff
@@ -0,0 +1,168 @@
+import {createContext, useReducer} from "react";
+import {initializeRecords} from "../utils";
+
+export const RecordContext = createContext(null);
+export const RecordDispatchContext = createContext(null);
+
+export const RecordProvider = ({children}) => {
+	const initialData = initializeRecords();
+	const [state, dispatch] = useReducer(recordReducer, initialData);
+
+	return (
+		<RecordContext.Provider value={state}>
+			<RecordDispatchContext.Provider value={dispatch}>
+				{children}
+			</RecordDispatchContext.Provider>
+		</RecordContext.Provider>
+	);
+};
+
+/* action type */
+const ADD_ITEM = 'recordReducer/ADDITEM';
+const CHANGE_DATASIZE = 'recordReducer/CHANGEDATASIZE';
+const CHANGE_MIN_HEIGHT = 'recordReducer/CHANGEMINHEIGHT';
+const CHANGE_MIN_WIDTH = 'recordReducer/CHANGEMINWIDTH';
+const CHANGE_SPACING = 'recordReducer/CHANGESPACING';
+const DELETE_ITEM = 'recordReducer/DELETEITEM';
+const DELETE_SELECTED_ITEM = 'recordReducer/DELETESELECTEDITEM';
+const SELECT_ALL = 'recordReducer/SELECTALL';
+const SELECT_ITEM = 'recordReducer/SELECTITEM';
+const SELECTION_ENABLE = 'recordReducer/SELECTIONENABLE';
+const SET_DATA = 'recordReducer/SETDATA';
```

Comment:
Is there a guide or rule to make these action type name?
If not, I prefer the same string as the variable name like `ADD_ITEM`.

Reply:
Redux uses uppercase camel case, and users seem to use that convention as a convention. and It seems React uses it in another way. So, it's up to us to decide, but i follow conventional rules anyway

https://redux.js.org/style-guide/#write-action-types-as-domaineventname
https://react.dev/reference/react/useReducer

Reply:
👍 Let's figure out which case is dominant recently.

Reply:
We decided to go as is.